File: medplum-java/medplum-model/src/main/java/com/fhir/MedicationRequest_DispenseRequest.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicationRequest_DispenseRequest.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicationRequest_DispenseRequest {
  Optional<MedicationRequest_InitialFill> initialFill();

  Optional<Reference> performer();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<Quantity> quantity();

  Optional<Duration> dispenseInterval();

  Optional<Period> validityPeriod();

  Optional<unsignedInt> numberOfRepeatsAllowed();

  Optional<List<Extension>> extension();

  Optional<Duration> expectedSupplyDuration();

  static ImmutableMedicationRequest_DispenseRequest.Builder builder() {
    return ImmutableMedicationRequest_DispenseRequest.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/MedicationRequest_DispenseRequest.ts
import { Duration } from "./Duration";
import { Extension } from "./Extension";
import { MedicationRequest_InitialFill } from "./MedicationRequest_InitialFill";
import { Period } from "./Period";
import { Quantity } from "./Quantity";
import { Reference } from "./Reference";
import { unsignedInt } from "./unsignedInt";

export interface MedicationRequest_DispenseRequest {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    initialFill: MedicationRequest_InitialFill | null | undefined;
    dispenseInterval: Duration | null | undefined;
    validityPeriod: Period | null | undefined;
    numberOfRepeatsAllowed: unsignedInt | null | undefined;
    quantity: Quantity | null | undefined;
    expectedSupplyDuration: Duration | null | undefined;
    performer: Reference | null | undefined;
}
